test/gtest/possible-contents.cpp

```diff
@@ -0,0 +1,296 @@
+#include "ir/possible-contents.h"
+#include "wasm-s-parser.h"
+#include "wasm.h"
+#include "gtest/gtest.h"
+
+using namespace wasm;
+
+// Asserts a == b, in any order.
+template<typename T> void assertEqualSymmetric(const T& a, const T& b) {
+  std::cout << "\nassertEqualSymmetric\n";
+  a.dump(std::cout);
+  std::cout << '\n';
+  b.dump(std::cout);
+  std::cout << '\n';
+
+  EXPECT_TRUE(a == b);
+  EXPECT_TRUE(b == a);
+  EXPECT_FALSE(a != b);
+  EXPECT_FALSE(b != a);
+}
+
+// Asserts a != b, in any order.
+template<typename T> void assertNotEqualSymmetric(const T& a, const T& b) {
+  std::cout << "\nassertNotEqualSymmetric\n";
+  a.dump(std::cout);
+  std::cout << '\n';
+  b.dump(std::cout);
+  std::cout << '\n';
+
+  EXPECT_TRUE(a != b);
+  EXPECT_TRUE(b != a);
+  EXPECT_FALSE(a == b);
+  EXPECT_FALSE(b == a);
+}
+
+// Asserts a combined with b (in any order) is equal to c.
+template<typename T>
+void assertCombination(const T& a, const T& b, const T& c) {
+  std::cout << "\nassertCombination\n";
+  a.dump(std::cout);
+  std::cout << '\n';
+  b.dump(std::cout);
+  std::cout << '\n';
+  c.dump(std::cout);
+  std::cout << '\n';
+
+  T temp1 = a;
+  temp1.combine(b);
+  temp1.dump(std::cout);
+  std::cout << '\n';
+  assertEqualSymmetric(temp1, c);
+
+  T temp2 = b;
+  temp2.combine(a);
+  temp2.dump(std::cout);
+  std::cout << '\n';
+  assertEqualSymmetric(temp2, c);
+}
+
+// Parse a module from text and return it.
+static std::unique_ptr<Module> parse(std::string module) {
+  auto wasm = std::make_unique<Module>();
+  wasm->features = FeatureSet::All;
+  try {
+    SExpressionParser parser(&module.front());
+    Element& root = *parser.root;
+    SExpressionWasmBuilder builder(*wasm, *root[0], IRProfile::Normal);
+  } catch (ParseException& p) {
+    p.dump(std::cerr);
+    Fatal() << "error in parsing wasm text";
+  }
+  return wasm;
+};
+
+// We want to declare a bunch of globals that are used in various tests. Doing
+// so in the global scope is risky, as their constructors use types, and the
+// type system itself uses global constructors. Instead, put all the tests
+// inside a "testMain()" function whose scope will contain those reused
+// "globals."
```

Comment:
This sounds like a good job for a test fixture. Docs are here: https://google.github.io/googletest/primer.html#same-data-multiple-tests and you can copy the structure of our type test fixture here: https://github.com/WebAssembly/binaryen/blob/main/test/gtest/type-test.h

Reply:
Thanks, refactored now to use a fixture.